catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeImpl.java

```diff
@@ -114,12 +163,24 @@ public String getProperty() {
     if (!isLeaf()) {
       throw new IllegalStateException("No property value exists for a logical operator");
     }
-    return property;
+    if (!(property instanceof String)) {
```

Comment:
❓  will `property` ever be null? or something besides a string or function?

Reply:
The `instanceof` check will return `false` if `property` is `null` and it'll only ever been `null` if the UI sends it that way.

Reply:
I use the alternate method, `getPropertyFunction` which doesn't fail that way, to check if it's a function. Otherwise we call this. If the UI does happen to send us a `null` node the whole request will fail gracefully since forms let exceptions propagate from the conversion up to the request level.